charts/agent/templates/configmap.yaml

```diff
@@ -37,14 +37,21 @@ data:
   Checking here the user is using Custom CA and if http_proxy.ssl = true
   If these conditions are true, then we use the agent.sslCaFileName for the http_proxy.ca_certificate
 */}}
-{{- if and (eq (include "sysdig.custom_ca.enabled"  (dict "global" .Values.global.ssl "component" .Values.ssl)) "true") (.Values.sysdig.settings) (hasKey .Values.sysdig.settings "http_proxy") (hasKey (default dict .Values.sysdig.settings.http_proxy) "ssl") (eq (get (default (dict "ssl" false) .Values.sysdig.settings.http_proxy) "ssl") true) }}
-  {{- $baseSettings := .Values.sysdig.settings -}}
-  {{- $caFilePath := printf "%s%s" "/etc/ca-certs/" (include "sysdig.custom_ca.keyName"  (dict "global" .Values.global.ssl "component" .Values.ssl)) }}
-  {{- $mergedSettings := mergeOverwrite $baseSettings (dict "http_proxy" (dict "ca_certificate" $caFilePath)) -}}
-  {{ toYaml $mergedSettings | nindent 4 }}
-{{- else if .Values.sysdig.settings }}
-  {{ toYaml .Values.sysdig.settings | nindent 4 }}
+{{- if and (eq (include "sysdig.custom_ca.enabled" (dict "global" .Values.global.ssl "component" .Values.ssl)) "true") }}
+  {{- if (hasKey .Values.sysdig.settings "http_proxy") }}
+    {{- if (hasKey .Values.sysdig.settings.http_proxy "ssl") }}
+      {{- if (.Values.sysdig.settings.http_proxy.ssl) }}
+        {{- $baseSettings := .Values.sysdig.settings -}}
+        {{- $caFilePath := printf "%s%s" "/etc/ca-certs/" (include "sysdig.custom_ca.keyName"  (dict "global" .Values.global.ssl "component" .Values.ssl)) }}
+        {{- $mergedSettings := mergeOverwrite $baseSettings (dict "http_proxy" (dict "ca_certificate" $caFilePath)) -}}
+        {{ toYaml $mergedSettings | nindent 4 }}
+      {{- end }}
+    {{- end }}
+  {{- end }}
 {{- end }}
+{{- if .Values.sysdig.settings }}
+  {{ toYaml .Values.sysdig.settings | nindent 4 }}
```

Comment:
This might result in some duplication since https://github.com/sysdiglabs/charts/pull/1524/files#diff-3cf829d749cba0e0b831af9d7d58cd1b70ff2039e2af38f06523ef121fa47a72R47 we are already returning also `.Values.sysdig.settings`